### DISCUSSION TITLE
Temporarily pause message consumption in consumers

### DISCUSSION THREAD

**Discussion by nadamgm** [rabbitmq/rabbitmq-dotnet-client#1496] [Q&A] [answered]
Hi everyone,

I'm trying to implement a sort of priority consumer, I'm trying to pause consumer A while consumer B still has messages in its queue.

I thought I'd implement this by using `IModel.BasicCancel(consumerTag)` and then calling `IModel.BasicConsume` again when the priority queue is empty.
I'm not sure this is a good way to do it though as I would imagine there could be unintended side-effects under the hood.
I don't want to close the channel while I'm suspending consumption, I just temporarily want to pause the consumption until Consumer B's queue is empty again (most of the time this should be the case the majority of the time with a few seconds of processing, however in peak load times, consumer B could take a few minutes to churn through its queue).

Is there a better way of doing this sort of stuff?

**Comment by lukebakken** [comment]
> I'm not sure this is a good way to do it though as I would imagine there could be unintended side-effects under the hood.

Why? Have you tried it out?

**Reply by nadamgm** [reply]
I'm about to try it. I'm just a bit unclear how it works under the hood. I haven't found much documentation on the topic.

**Reply by lukebakken** [reply]
Give it a try first. If it doesn't work as you expect, follow-up and provide a git repository with code I can compile and run to reproduce what you're doing.

**Reply by nadamgm** [reply, marked as answer]
I've tried it, it does seem to work using these methods.

**Reply by michaelklishin** [reply]
Cancelling a consumer to re-register it later works perfectly fine unless you do it very rapidly in sequence (e.g. multiple times a second). Even then it can work well but it will be very difficult to reason about your app.
